src/System.Private.Xml/src/System/Xml/Dom/XmlName.cs

```diff
@@ -182,18 +183,9 @@ public static int GetHashCode(string name)
             int hashCode = 0;
             if (name != null)
             {
-                unchecked
-                {
-                    for (int i = name.Length - 1; i >= 0; i--)
-                    {
-                        char ch = name[i];
-                        if (ch == ':') break;
-                        hashCode += (hashCode << 7) ^ ch;
-                    }
-                    hashCode -= hashCode >> 17;
-                    hashCode -= hashCode >> 11;
-                    hashCode -= hashCode >> 5;
-                }
+                string[] splitted = name.Split(":");
+                ReadOnlySpan<byte> bytes = MemoryMarshal.AsBytes(splitted[splitted.Length - 1].AsSpan());
```

Comment:
Hmm, isn't it always just `name.AsSpan(name.LastIndexOf(':') + 1)`?

If there's no colon, LastIndexOf returns -1, so +1 is 0 (the whole string).  If there is one, then it starts at the next character.

Reply:
yes, that is much better, thanks!